### DISCUSSION TITLE
Snapcast Server Connection to TCP Audio Stream

### DISCUSSION THREAD

**Discussion by dycfan** [snapcast/snapcast#1221] [General]
Hello!

I have successfully been streaming my audio to port 3266 of my TCP server. Now, I need to configure this audio stream in the Snapcast configuration file. I have tried the following methods:

Method One:
[stream]
source = tcp://192.168.9.6:3266?name=DesktopMusic&mode=client
sampleformat = 44100:16:2
codec = flac
chunk_ms = 20
buffer = 1000
send_to_muted = false

Method Two:
[stream]
source = tcp://192.168.9.6?name=DesktopMusic&mode=server
sampleformat = 44100:16:2
codec = flac
chunk_ms = 20
buffer = 1000
send_to_muted = false

Method Three:
[stream]
source = tcp://127.0.0.1:3266?name=DesktopMusic&mode=client
sampleformat = 44100:16:2
codec = flac
chunk_ms = 20
buffer = 1000
send_to_muted = false

Method Four:
[stream]
source = tcp://127.0.0.1?name=DesktopMusic&mode=server
sampleformat = 44100:16:2
codec = flac
chunk_ms = 20
buffer = 1000
send_to_muted = false

The IP address of the host running my Snapcast server is 192.168.9.6.

Unfortunately, none of the above attempts successfully input the audio stream sent to port 3266 of my local TCP server into Snapcast. At a loss, I'm reaching out to you for assistance!

I'm looking forward to your guidance and prompt response.

Thank you very much!

**Comment by oliv3r** [comment]
I think that I'm sort of asking for the same, 'what is the protocol' for the tcp server, and how to use it ...
https://github.com/badaix/snapcast/issues/1222

**Comment by SantiagoSotoC** [comment]
You have to use 0.0.0.0.0 as ip, at least that's the way I got it to work
